spec/javascript/packages/document-capture/components/acuant-capture-spec.jsx

```diff
@@ -1149,26 +1158,79 @@ describe('document-capture/components/acuant-capture', () => {
 
   it('logs metrics for manual upload', async () => {
     const trackEvent = sinon.stub();
+    const onChange = sinon.stub();
+
     const { getByLabelText } = render(
       <AnalyticsContext.Provider value={{ trackEvent }}>
-        <AcuantContextProvider sdkSrc="about:blank" cameraSrc="about:blank">
-          <AcuantCapture label="Image" name="test" />
-        </AcuantContextProvider>
+        <FailedCaptureAttemptsContextProvider
+          maxCaptureAttemptsBeforeNativeCamera={3}
+          maxSubmissionAttemptsBeforeNativeCamera={3}
+        >
+          <AcuantContextProvider sdkSrc="about:blank" cameraSrc="about:blank">
+            <AcuantCapture label="Image" name="front" onChange={onChange} />
+          </AcuantContextProvider>
+        </FailedCaptureAttemptsContextProvider>
       </AnalyticsContext.Provider>,
     );
-
     const input = getByLabelText('Image');
     uploadFile(input, validUpload);
+    onChange.calls;
+    await new Promise((resolve) => onChange.callsFake(resolve));
+    expect(trackEvent).to.be.calledOnce();
+    expect(trackEvent).to.have.been.calledWith(
+      'IdV: front image added',
+      sinon.match({
+        width: sinon.match.number,
+        height: sinon.match.number,
+        fingerprint: sinon.match.string,
+        source: 'upload',
+        mimeType: 'image/jpeg',
+        size: sinon.match.number,
+        attempt: sinon.match.number,
+        acuantCaptureMode: 'AUTO',
+      }),
+    );
+  });
 
-    await expect(trackEvent).to.eventually.be.calledWith('IdV: test image added', {
-      height: sinon.match.number,
-      mimeType: 'image/jpeg',
-      source: 'upload',
-      width: sinon.match.number,
-      attempt: sinon.match.number,
-      size: sinon.match.number,
-      acuantCaptureMode: sinon.match.string,
-    });
+  it('logs metrics for failed reupload', async () => {
+    const trackEvent = sinon.stub();
+    const onChange = sinon.stub();
+    class TestComponent extends PureComponent {
+      // eslint-disable-next-line react/static-property-placement
+      static contextType = FailedCaptureAttemptsContext;
+
+      render() {
+        this.context.failedSubmissionImageFingerprints = {
+          front: ['kgLjncfQAICyEYQhdFMAAKxdFceQ80WPjwK2puuuLd8'],
+          back: [],
+        };
```

Comment:
`render` shouldn't have side-effects like this. Can we instead wrap with a `<FailedCaptureAttemptsContextProvider>` and allow the fingerprints to be passed into the provider? Or update the test so that it fails once to get the fingerprint into the context, and then makes a second attempt to verify that the expected event is logged?

Reply:
@aduth, fixed by passing into the context provider.